### ISSUE TITLE
3.5beta1 - Permission editing is broken

### ISSUE THREAD

**Issue by Fedik** [joomla/joomla-cms#8321]
I tried test the beta, and noticed that permission editing does not really work, or works randomly.
After changing any permission dropdown I got undefined error.

**Global configuration:**
After any change I got undefined error and X icon
![screen 2015-11-07 15 36 14 1241x383](https://cloud.githubusercontent.com/assets/1568198/11015386/d87ddc46-8565-11e5-8f95-119ed9204d6c.png)

**Module editing:**
Sometimes it works, sometimes it is not.
![screen 2015-11-07 15 37 09 836x433](https://cloud.githubusercontent.com/assets/1568198/11015389/e700dc96-8565-11e5-97fc-585e901bfebe.png)

**Comment by Fedik** [comment]
it is after #8132
@tflm84 can you please check?

**Comment by zero-24** [comment]
Hmm i have no Problem. Can You make sure You have cleaned The browser cache?

**Comment by Fedik** [comment]
yes, and I made clear installation for J3.5Beta1 
strange

**Comment by Fedik** [comment]
ok, 
database driver MySQL PDO => nothing works
MySQLi => sems works fine

**Comment by zero-24** [comment]
Oh yes that looks bad. I Can confirm it with pdo driver

**Comment by Fedik** [comment]
server answer does not looks as correct
```
{
"success":true,
"message":null,
"messages":null,
"data":"{\"queryString\":\"UPDATE `xxx_assets`\\nSET rules = '{\\\\\\\"core.login.site\\\\\\\":{\\\\\\\"1\\\\\\\":1,\\\\\\\"12\\\\\\\":1},\\\\\\\"core.login.offline\\\\\\\":{\\\\\\\"1\\\\\\\":1,\\\\\\\"12\\\\\\\":1},\\\\\\\"core.create\\\\\\\":{\\\\\\\"12\\\\\\\":1},\\\\\\\"core.login.admin\\\\\\\":{\\\\\\\"12\\\\\\\":0},\\\\\\\"core.admin\\\\\\\":{\\\\\\\"12\\\\\\\":1}}'\\nWHERE `name` = 'com_config'\"}"
}
```

**Comment by Fedik** [comment]
the thing that `$result = $this->db->execute();` for PDO the `$result` is some PDO object not a `boolean`

**Comment by zero-24** [comment]
I have a fix. Give ne a second to send zhe PR ;)

**Comment by zero-24** [comment]
Can you test #8322 @Fedik

**Comment by Fedik** [comment]
@zero-24 after #8322 the error disappeared :wink:  
but it still save nothing, after refresh the page I get old permissions.

try in Global configuration:
- Open the "Author" user group
- Set "Edit State" to "Allow" - you get "success" icon
- Push F5 to refresh the page
- Again, open the "Author" user group 

For "Edit State"  I get old "inherit" instead of "allow".

**Comment by zero-24** [comment]
it is the same behavior as on mysql ;) You need to hit the save button. The ajax efects just show how it will look like after the save.

**Comment by Fedik** [comment]
hm, nothing sorry .. still nothing saved 
strange

but as I see `ConfigModelApplication::storePermissions()` do some saving, now I confused :wink:

**Comment by zero-24** [comment]
Yes. I Can confirm nothing gets saved also not on hit the save button. :(

**Comment by Fedik** [comment]
form #8132 description:
>Now, the permissions will be stored immediately after changing a value via AJAX. This solution leads to smaller forms and consistent storage of the permissions.

**Comment by brianteeman** [comment]
If that's true then the description us wrong

**Comment by zero-24** [comment]
correct @brianteeman you can blame me i have missunderstand it (btw also save don't work) @Fedik is correct that it is broken

@dgt41 and I is working on a fix. Thanks

**Comment by roland-d** [comment]
Permission saving is fixed in #8343 this does **not** include the PDO fix.

**Comment by roland-d** [comment]
Closing this as we have a PR at #8343<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8321">issues.joomla.org/joomla-cms/8321</a>.</sub>
